src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/RecommendCaseInsensitiveStringComparison.Analyzer.cs

```diff
@@ -190,49 +205,165 @@ private void AnalyzeCompilationStart(CompilationStartAnalysisContext context)
 
             context.RegisterOperationAction(context =>
             {
-                IInvocationOperation caseChangingInvocation = (IInvocationOperation)context.Operation;
-                IMethodSymbol caseChangingMethod = caseChangingInvocation.TargetMethod;
-
-                if (!caseChangingMethod.Equals(toLowerParameterlessMethod) &&
-                    !caseChangingMethod.Equals(toLowerInvariantParameterlessMethod) &&
-                    !caseChangingMethod.Equals(toUpperParameterlessMethod) &&
-                    !caseChangingMethod.Equals(toUpperInvariantParameterlessMethod))
+                if (context.Operation is IInvocationOperation caseChangingInvocation)
```

Comment:
Nit: Alternatively, you can just register separate operation callbacks for each operation kind - up to you.

Reply:
@carlossanlop 

```csharp
context.RegisterOpationAction(context =>
{
    var caseChangingInvocation = (IInvocationOperation)context.Operation;
    AnalyzeInvocation(context, caseChangingInvocation, stringType,
        containsStringMethod, startsWithStringMethod, compareToStringMethod,
        indexOfStringMethod, indexOfStringInt32Method, indexOfStringInt32Int32Method);
}, OperationKind.Invocation);

context.RegisterOpationAction(context =>
{
    var binaryOperation = (IBinaryOperation)context.Operation;
    AnalyzeBinaryOperation(context, binaryOperation, stringType);
}, OperationKind.Binary);
```

Reply:
Thanks! I was doing something like that, but don't remember exactly what I was doing differently, it just wouldn't work. Let me test it again.